pkgs/by-name/ls/lsp-ai/package.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  cmake,
+  openssl,
+  zlib,
+  perl,
+}:
+
+let
+  pname = "lsp-ai";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "SilasMarvin";
+    repo = pname;
+    rev = "66655f6285fe8aad8f2d72646e9ed47c06245525";
+    hash = "sha256-DwqqZBzLevuRCW6QzGyWdE+JtpW6b3EMDuiWtajv/U4=";
+  };
+
+in
+rustPlatform.buildRustPackage {
+  inherit pname version src;
+
+  cargoLock = {
+    lockFile = src + "/Cargo.lock";
```

Comment:
This won't work. It's an Input From Derivation (IFD) because the content of Cargo.lock is parsed in Nix code and the crates are converted to individual derivations. If there are `git+` in Cargo.lock you have to also upload it.

Reply:
See https://nixos.org/manual/nixpkgs/unstable/#vendoring-of-dependencies